presto-main/src/main/java/com/facebook/presto/metadata/FunctionSignatureMatcher.java

```diff
@@ -110,6 +110,11 @@ private Optional<Signature> matchFunction(Collection<? extends SqlFunction> cand
             return Optional.of(getOnlyElement(applicableFunctions).getBoundSignature());
         }
 
+        //Hash_counts
+        if (applicableFunctions.size() == 2 && applicableFunctions.get(1).getBoundSignature().getName().getObjectName().equals("hash_counts")) {
```

Comment:
While this will make finding the solution harder, I don't think statically checking for a particular function inside `FunctionSignatureMatcher` is the way to go.  It needs to be generic because functions may change in the future.

Reply:
@tdcmeehan I ll check this again and revert back